projects/plugins/jetpack/extensions/shared/external-media/sources/openverse.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useRef, useCallback, useState, useEffect } from '@wordpress/element';
+import { TextControl, Button } from '@wordpress/components';
+import { sample } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { SOURCE_OPENVERSE, PEXELS_EXAMPLE_QUERIES } from '../constants';
+import withMedia from './with-media';
+import MediaBrowser from '../media-browser';
+import { getApiUrl } from './api';
+
+function OpenverseMedia( props ) {
+	const { media, isCopying, isLoading, pageHandle, multiple, copyMedia, getMedia } = props;
+
+	const [ searchQuery, setSearchQuery ] = useState( sample( PEXELS_EXAMPLE_QUERIES ) );
+	const [ lastSearchQuery, setLastSearchQuery ] = useState( '' );
+
+	const onCopy = useCallback(
+		items => {
+			copyMedia( items, getApiUrl( 'copy', SOURCE_OPENVERSE ), SOURCE_OPENVERSE );
+		},
+		[ copyMedia ]
+	);
+
+	const getNextPage = useCallback(
+		( event, reset = false ) => {
+			if ( searchQuery ) {
+				getMedia(
+					getApiUrl( 'list', SOURCE_OPENVERSE, {
+						number: 20,
+						search: searchQuery,
+					} ),
+					reset
+				);
+			}
+		},
+		[ getMedia, searchQuery ]
+	);
+
+	const previousSearchQueryValue = useRef();
+	const onSearch = useCallback(
+		event => {
+			event.preventDefault();
+			setLastSearchQuery( searchQuery );
+			getNextPage( event, true );
+			previousSearchQueryValue.current = searchQuery;
+		},
+		[ getNextPage, searchQuery ]
+	);
+
+	// Load initial results for the random example query. Only do it once.
+	useEffect( getNextPage, [] ); // eslint-disable-line react-hooks/exhaustive-deps
+
+	const searchFormEl = useRef( null );
+
+	const focusSearchInput = () => {
+		if ( ! searchFormEl.current ) {
+			return;
+		}
+
+		const formElements = Array.from( searchFormEl.current.elements );
+		// TextControl does not support ref forwarding, so we need to find the input:
+		const searchInputEl = formElements.find( element => element.type === 'search' );
+
+		if ( searchInputEl ) {
+			searchInputEl.focus();
+			searchInputEl.select();
+		}
+	};
+
+	useEffect( focusSearchInput, [] );
```

Comment:
defining the function and calling it once doesn't seem to be needed. Should we simply define the function straight here?